### PR TITLE
Do not convert IndexPropVar to direct if the expression contains indirects

### PR DESCRIPTION
**Pull request by isuruf**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #123661
* __->__ #130961

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/index_propagation.py [hunk 1]
```diff
@@ -324,29 +325,30 @@ def indirect_indexing(
...
-                -size <= expr
-            )
-            can_prove_upper = self.statically_true(expr < size)
-            expr = wrap_expr(expr)
-            if generate_assert(check):
-                self.fallback(
-                    "check_bounds",
-                    (expr, size),
-                    dict(lower=not can_prove_lower, upper=not can_prove_upper),
+            if not free_symbol_is_type(expr, SymT.INDIRECT):
```
Comment (peterbell10) [inline torch/_inductor/index_propagation.py:328]: Can you explain this? I don't see any reason why indirect indexing variable wouldn't work in a direct indexing expression.

Comment (isuruf) [reply torch/_inductor/index_propagation.py:328]: There's a test in the other PR in the stack that fails with

```
2024-07-16T19:17:48.7363794Z _ TestInductorOpInfoCUDA.test_comprehensive_nn_functional_unfold_cuda_float32 __
2024-07-16T19:17:48.7364908Z Traceback (most recent call last):
2024-07-16T19:17:48.7366475Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_device_type.py", line 1127, in test_wrapper
2024-07-16T19:17:48.7367906Z     return test(*args, **kwargs)
2024-07-16T19:17:48.7369357Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_device_type.py", line 1390, in only_fn
2024-07-16T19:17:48.7370541Z     return fn(self, *args, **kwargs)
2024-07-16T19:17:48.7371827Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 2005, in wrapper
2024-07-16T19:17:48.7373116Z     fn(*args, **kwargs)
2024-07-16T19:17:48.7374483Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_device_type.py", line 1197, in dep_fn
2024-07-16T19:17:48.7375888Z     return fn(slf, *args, **kwargs)
2024-07-16T19:17:48.7377395Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_device_type.py", line 1197, in dep_fn
2024-07-16T19:17:48.7378694Z     return fn(slf, *args, **kwargs)
2024-07-16T19:17:48.7380207Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 1428, in wrapper
2024-07-16T19:17:48.7381519Z     fn(*args, **kwargs)
2024-07-16T19:17:48.7382877Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 1377, in wrapper
2024-07-16T19:17:48.7384095Z     fn(*args, **kwargs)
2024-07-16T19:17:48.7384978Z   File "/opt/conda/envs/py_3.10/lib/python3.10/unittest/mock.py", line 1379, in patched
2024-07-16T19:17:48.7386036Z     return func(*newargs, **newkeywargs)
2024-07-16T19:17:48.7387031Z   File "/opt/conda/envs/py_3.10/lib/python3.10/contextlib.py", line 79, in inner
2024-07-16T19:17:48.7388019Z     return func(*args, **kwds)
2024-07-16T19:17:48.7389106Z   File "/var/lib/jenkins/workspace/test/inductor/test_torchinductor_opinfo.py", line 548, in inner
2024-07-16T19:17:48.7390225Z     raise e
2024-07-16T19:17:48.7391228Z   File "/var/lib/jenkins/workspace/test/inductor/test_torchinductor_opinfo.py", line 540, in inner
2024-07-16T19:17:48.7392366Z     fn(self, device, dtype, op)
2024-07-16T19:17:48.7393619Z   File "/var/lib/jenkins/workspace/test/inductor/test_torchinductor_opinfo.py", line 784, in test_comprehensive
2024-07-16T19:17:48.7394846Z     raise e
2024-07-16T19:17:48.7395908Z   File "/var/lib/jenkins/workspace/test/inductor/test_torchinductor_opinfo.py", line 744, in test_comprehensive
2024-07-16T19:17:48.7397138Z     self.check_model_gpu(
2024-07-16T19:17:48.7398027Z   File "/opt/conda/envs/py_3.10/lib/python3.10/contextlib.py", line 79, in inner
2024-07-16T19:17:48.7399191Z     return func(*args, **kwds)
2024-07-16T19:17:48.7400312Z   File "/var/lib/jenkins/workspace/test/inductor/test_torchinductor.py", line 599, in check_model_gpu
2024-07-16T19:17:48.7401469Z     check_model(
2024-07-16T19:17:48.7402470Z   File "/var/lib/jenkins/workspace/test/inductor/test_torchinductor.py", line 446, in check_model
2024-07-16T19:17:48.7403667Z     actual = run(*example_inputs, **kwargs)
2024-07-16T19:17:48.7405071Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 435, in _fn
2024-07-16T19:17:48.7406248Z     return fn(*args, **kwargs)
2024-07-16T19:17:48.7407585Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 1121, in __call__
2024-07-16T19:17:48.7408895Z     return self._torchdynamo_orig_callable(
2024-07-16T19:17:48.7410361Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 472, in __call__
2024-07-16T19:17:48.7411590Z     return _compile(
2024-07-16T19:17:48.7412422Z   File "/opt/conda/envs/py_3.10/lib/python3.10/contextlib.py", line 79, in inner
2024-07-16T19:17:48.7413556Z     return func(*args, **kwds)
2024-07-16T19:17:48.7414820Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 817, in _compile
2024-07-16T19:17:48.7416328Z     guarded_code = compile_inner(code, one_graph, hooks, transform)
2024-07-16T19:17:48.7417869Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/utils.py", line 240, in time_wrapper
2024-07-16T19:17:48.7419186Z     r = func(*args, **kwargs)
2024-07-16T19:17:48.7420542Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_utils_internal.py", line 85, in wrapper_function
2024-07-16T19:17:48.7422040Z     return StrobelightCompileTimeProfiler.profile_compile_time(
2024-07-16T19:17:48.7423891Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_strobelight/compile_time_profiler.py", line 129, in profile_compile_time
2024-07-16T19:17:48.7425376Z     return func(*args, **kwargs)
2024-07-16T19:17:48.7426799Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 636, in compile_inner
2024-07-16T19:17:48.7428215Z     out_code = transform_code_object(code, transform)
2024-07-16T19:17:48.7429892Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/bytecode_transformation.py", line 1280, in transform_code_object
2024-07-16T19:17:48.7431493Z     transformations(instructions, code_options)
2024-07-16T19:17:48.7432902Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 178, in _fn
2024-07-16T19:17:48.7434041Z     return fn(*args, **kwargs)
2024-07-16T19:17:48.7435345Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 581, in transform
2024-07-16T19:17:48.7436549Z     tracer.run()
2024-07-16T19:17:48.7437770Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 2498, in run
2024-07-16T19:17:48.7439170Z     super().run()
2024-07-16T19:17:48.7440377Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 908, in run
2024-07-16T19:17:48.7441607Z     while self.step():
2024-07-16T19:17:48.7442866Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 820, in step
2024-07-16T19:17:48.7444176Z     self.dispatch_table[inst.opcode](self, inst)
2024-07-16T19:17:48.7445712Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 2689, in RETURN_VALUE
2024-07-16T19:17:48.7446992Z     self._return(inst)
2024-07-16T19:17:48.7448296Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 2674, in _return
2024-07-16T19:17:48.7449605Z     self.output.compile_subgraph(
2024-07-16T19:17:48.7450997Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/output_graph.py", line 1102, in compile_subgraph
2024-07-16T19:17:48.7452513Z     self.compile_and_call_fx_graph(tx, list(reversed(stack_values)), root)
2024-07-16T19:17:48.7453741Z   File "/opt/conda/envs/py_3.10/lib/python3.10/contextlib.py", line 79, in inner
2024-07-16T19:17:48.7454715Z     return func(*args, **kwds)
2024-07-16T19:17:48.7456142Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/output_graph.py", line 1324, in compile_and_call_fx_graph
2024-07-16T19:17:48.7457614Z     compiled_fn = self.call_user_compiler(gm)
2024-07-16T19:17:48.7459119Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/utils.py", line 240, in time_wrapper
2024-07-16T19:17:48.7460296Z     r = func(*args, **kwargs)
2024-07-16T19:17:48.7461717Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/output_graph.py", line 1415, in call_user_compiler
2024-07-16T19:17:48.7463279Z     raise BackendCompilerFailed(self.compiler_fn, e).with_traceback(
2024-07-16T19:17:48.7464997Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/output_graph.py", line 1396, in call_user_compiler
2024-07-16T19:17:48.7466681Z     compiled_fn = compiler_fn(gm, self.example_inputs())
2024-07-16T19:17:48.7468373Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/repro/after_dynamo.py", line 129, in __call__
2024-07-16T19:17:48.7469763Z     compiled_gm = compiler_fn(gm, example_inputs)
2024-07-16T19:17:48.7471312Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/repro/after_dynamo.py", line 129, in __call__
2024-07-16T19:17:48.7472704Z     compiled_gm = compiler_fn(gm, example_inputs)
2024-07-16T19:17:48.7474005Z   File "/var/lib/jenkins/workspace/test/inductor/test_torchinductor.py", line 438, in compile_fx_wrapper
2024-07-16T19:17:48.7475298Z     return compile_fx(model_, example_inputs_)
2024-07-16T19:17:48.7476327Z   File "/opt/conda/envs/py_3.10/lib/python3.10/contextlib.py", line 79, in inner
2024-07-16T19:17:48.7477287Z     return func(*args, **kwds)
2024-07-16T19:17:48.7478863Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/compile_fx.py", line 1547, in compile_fx
2024-07-16T19:17:48.7480132Z     return aot_autograd(
2024-07-16T19:17:48.7481479Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/backends/common.py", line 69, in __call__
2024-07-16T19:17:48.7482928Z     cg = aot_module_simplified(gm, example_inputs, **self.kwargs)
2024-07-16T19:17:48.7484636Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py", line 995, in aot_module_simplified
2024-07-16T19:17:48.7486039Z     compiled_fn = dispatch_and_compile()
2024-07-16T19:17:48.7487606Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py", line 984, in dispatch_and_compile
2024-07-16T19:17:48.7489028Z     compiled_fn, _ = create_aot_dispatcher_function(
2024-07-16T19:17:48.7490494Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/utils.py", line 240, in time_wrapper
2024-07-16T19:17:48.7491689Z     r = func(*args, **kwargs)
2024-07-16T19:17:48.7493228Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py", line 713, in create_aot_dispatcher_function
2024-07-16T19:17:48.7494734Z     compiled_fn, fw_metadata = compiler_fn(
2024-07-16T19:17:48.7496567Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_functorch/_aot_autograd/jit_compile_runtime_wrappers.py", line 489, in aot_dispatch_autograd
2024-07-16T19:17:48.7498402Z     compiled_fw_func = aot_config.fw_compiler(fw_module, adjusted_flat_args)
2024-07-16T19:17:48.7500138Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/utils.py", line 240, in time_wrapper
2024-07-16T19:17:48.7501367Z     r = func(*args, **kwargs)
2024-07-16T19:17:48.7502772Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/compile_fx.py", line 1453, in fw_compiler_base
2024-07-16T19:17:48.7504079Z     return inner_compile(
2024-07-16T19:17:48.7505422Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/repro/after_aot.py", line 84, in debug_wrapper
2024-07-16T19:17:48.7506858Z     inner_compiled_fn = compiler_fn(gm, example_inputs)
2024-07-16T19:17:48.7508323Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/debug.py", line 304, in inner
2024-07-16T19:17:48.7509517Z     return fn(*args, **kwargs)
2024-07-16T19:17:48.7510438Z   File "/opt/conda/envs/py_3.10/lib/python3.10/contextlib.py", line 79, in inner
2024-07-16T19:17:48.7511400Z     return func(*args, **kwds)
2024-07-16T19:17:48.7512304Z   File "/opt/conda/envs/py_3.10/lib/python3.10/contextlib.py", line 79, in inner
2024-07-16T19:17:48.7513271Z     return func(*args, **kwds)
2024-07-16T19:17:48.7514589Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/compile_fx.py", line 431, in wrapper
2024-07-16T19:17:48.7515833Z     return fn(*args, **kwargs)
2024-07-16T19:17:48.7517138Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/utils.py", line 240, in time_wrapper
2024-07-16T19:17:48.7518967Z     r = func(*args, **kwargs)
2024-07-16T19:17:48.7520100Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/compile_fx.py", line 534, in compile_fx_inner
2024-07-16T19:17:48.7520988Z     compiled_graph = FxGraphCache.load(
2024-07-16T19:17:48.7521888Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/codecache.py", line 1107, in load
2024-07-16T19:17:48.7522801Z     compiled_graph = compile_fx_fn(gm, example_inputs, **fx_kwargs)
2024-07-16T19:17:48.7523563Z   File "/opt/conda/envs/py_3.10/lib/python3.10/contextlib.py", line 79, in inner
2024-07-16T19:17:48.7524190Z     return func(*args, **kwds)
2024-07-16T19:17:48.7525125Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/compile_fx.py", line 866, in fx_codegen_and_compile
2024-07-16T19:17:48.7526017Z     compiled_fn = graph.compile_to_fn()
2024-07-16T19:17:48.7526918Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/graph.py", line 1749, in compile_to_fn
2024-07-16T19:17:48.7527742Z     return self.compile_to_module().call
2024-07-16T19:17:48.7528629Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/utils.py", line 240, in time_wrapper
2024-07-16T19:17:48.7529411Z     r = func(*args, **kwargs)
2024-07-16T19:17:48.7530283Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/graph.py", line 1677, in compile_to_module
2024-07-16T19:17:48.7531282Z     self.codegen_with_cpp_wrapper() if self.cpp_wrapper else self.codegen()
2024-07-16T19:17:48.7532308Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/graph.py", line 1628, in codegen
2024-07-16T19:17:48.7533128Z     self.scheduler = Scheduler(self.buffers)
2024-07-16T19:17:48.7534028Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/utils.py", line 240, in time_wrapper
2024-07-16T19:17:48.7534802Z     r = func(*args, **kwargs)
2024-07-16T19:17:48.7535655Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/scheduler.py", line 1451, in __init__
2024-07-16T19:17:48.7536581Z     self.nodes = [self.create_scheduler_node(n) for n in nodes]
2024-07-16T19:17:48.7537590Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/scheduler.py", line 1451, in <listcomp>
2024-07-16T19:17:48.7538510Z     self.nodes = [self.create_scheduler_node(n) for n in nodes]
2024-07-16T19:17:48.7539782Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/scheduler.py", line 1550, in create_scheduler_node
2024-07-16T19:17:48.7540707Z     return SchedulerNode(self, node)
2024-07-16T19:17:48.7541586Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/scheduler.py", line 751, in __init__
2024-07-16T19:17:48.7542369Z     self._compute_attrs()
2024-07-16T19:17:48.7543209Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/scheduler.py", line 758, in _compute_attrs
2024-07-16T19:17:48.7544138Z     self._sizes, self._body = self.node.simplify_and_reorder(
2024-07-16T19:17:48.7545156Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/ir.py", line 3543, in simplify_and_reorder
2024-07-16T19:17:48.7545990Z     ) = self.get_default_sizes_body()
2024-07-16T19:17:48.7546829Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/utils.py", line 442, in wrapper
2024-07-16T19:17:48.7547597Z     setattr(self, key, fn(self))
2024-07-16T19:17:48.7548492Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/ir.py", line 3500, in get_default_sizes_body
2024-07-16T19:17:48.7549306Z     body = LoopBody(
2024-07-16T19:17:48.7550059Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/ir.py", line 6489, in __init__
2024-07-16T19:17:48.7550884Z     self.root_block = LoopBodyBlock(self, fn, args)
2024-07-16T19:17:48.7551786Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/ir.py", line 6727, in __init__
2024-07-16T19:17:48.7552628Z     ops.output(fn(*args))
2024-07-16T19:17:48.7553484Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/ir.py", line 517, in store_output
2024-07-16T19:17:48.7554380Z     return ops.store(output_name, indexer(vars), loader(vars))
2024-07-16T19:17:48.7555327Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/ir.py", line 2027, in loader
2024-07-16T19:17:48.7556081Z     return inner(reindex(idx))
2024-07-16T19:17:48.7556926Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/lowering.py", line 3010, in inner_fn
2024-07-16T19:17:48.7557736Z     return x_loader(index_inner_fn(idx))
2024-07-16T19:17:48.7558889Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/lowering.py", line 3194, in inner_fn
2024-07-16T19:17:48.7559885Z     return ops.masked(mask_val, lambda: self_loader(_unsafe_index_fn(idx)), fill)
2024-07-16T19:17:48.7560960Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/virtualized.py", line 262, in inner
2024-07-16T19:17:48.7561911Z     return OpsWrapper._wrap(getattr(_ops, name)(*new_args, **new_kwargs))
2024-07-16T19:17:48.7562980Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/index_propagation.py", line 282, in inner
2024-07-16T19:17:48.7563849Z     return self.fallback(name, args, kwargs)
2024-07-16T19:17:48.7564842Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/index_propagation.py", line 255, in fallback
2024-07-16T19:17:48.7565842Z     return self.wrap(getattr(self._inner, name)(*new_args, **new_kwargs))
2024-07-16T19:17:48.7566831Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/ir.py", line 6648, in masked
2024-07-16T19:17:48.7567691Z     subblock = LoopBodyBlock(self.body, masked_body, [])
2024-07-16T19:17:48.7568626Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/ir.py", line 6727, in __init__
2024-07-16T19:17:48.7569405Z     ops.output(fn(*args))
2024-07-16T19:17:48.7572180Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/lowering.py", line 3194, in <lambda>
2024-07-16T19:17:48.7573198Z     return ops.masked(mask_val, lambda: self_loader(_unsafe_index_fn(idx)), fill)
2024-07-16T19:17:48.7574262Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/lowering.py", line 2953, in fn
2024-07-16T19:17:48.7575040Z     ops.indirect_indexing(
2024-07-16T19:17:48.7575938Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/virtualized.py", line 284, in indirect_indexing
2024-07-16T19:17:48.7576861Z     return _ops.indirect_indexing(index, size, check)
2024-07-16T19:17:48.7577909Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/index_propagation.py", line 338, in indirect_indexing
2024-07-16T19:17:48.7579134Z     can_prove_lower = self.statically_true(0 <= expr) or self.statically_true(
2024-07-16T19:17:48.7580302Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/index_propagation.py", line 309, in statically_true
2024-07-16T19:17:48.7581258Z     evaluated = self.shape_env._maybe_evaluate_static(
2024-07-16T19:17:48.7582294Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/fx/experimental/symbolic_shapes.py", line 1458, in wrapper
2024-07-16T19:17:48.7583187Z     return fn_cache(self, *args, **kwargs)
2024-07-16T19:17:48.7584252Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/fx/experimental/symbolic_shapes.py", line 4496, in _maybe_evaluate_static
2024-07-16T19:17:48.7585194Z     vr = var_ranges[k]
2024-07-16T19:17:48.7585801Z torch._dynamo.exc.BackendCompilerFailed: backend='compile_fx_wrapper' raised:
2024-07-16T19:17:48.7586421Z KeyError: indirect0
2024-07-16T19:17:48.7586625Z 
```

https://hud.pytorch.org/pytorch/pytorch/pull/123661?sha=4150bee1ad638e2c84cd8b807a756acaaaf1b9c8

My initial thought was that `self.var_to_range` was that this should have hit the else codepath in `indirect_indexing`.

Any ideas on the error in the log?

Comment (peterbell10) [reply torch/_inductor/index_propagation.py:328]: Looks like #127677, did you have #128378 in your history when you saw this?

Comment (isuruf) [reply torch/_inductor/index_propagation.py:328]: Just checked with latest main branch and I can still see the error.

Comment (peterbell10) [reply torch/_inductor/index_propagation.py:328]: cc @ColinPeppler

Comment (isuruf) [reply torch/_inductor/index_propagation.py:328]: If I revert the changes in `torch/fx/experimental/symbolic_shapes.py` from #128378, the test pass with warnings.

Comment (peterbell10) [reply torch/_inductor/index_propagation.py:328]: I have a fix in #130984

Comment (isuruf) [reply torch/_inductor/index_propagation.py:328]: Thanks. I've confirmed that it works.

**Pull request closed without merging**